src/merkle/partial_mt/tests.rs

```diff
@@ -0,0 +1,128 @@
+use super::{
+    super::{int_to_node, MerkleStore, MerkleTree, NodeIndex, PartialMerkleTree},
+    Word,
+};
+
+// TEST DATA
+// ================================================================================================
+
+const NODE10: NodeIndex = NodeIndex::new_unchecked(1, 0);
+
+const NODE22: NodeIndex = NodeIndex::new_unchecked(2, 2);
+
+const NODE32: NodeIndex = NodeIndex::new_unchecked(3, 2);
+const NODE33: NodeIndex = NodeIndex::new_unchecked(3, 3);
+
+const VALUES8: [Word; 8] = [
+    int_to_node(1),
+    int_to_node(2),
+    int_to_node(3),
+    int_to_node(4),
+    int_to_node(5),
+    int_to_node(6),
+    int_to_node(7),
+    int_to_node(8),
+];
+
+// TESTS
+// ================================================================================================
+
+// with_paths CONSTRUCTOR TESTS
+// ------------------------------------------------------------------------------------------------
+
+#[test]
+fn get_root() {
+    let mt = MerkleTree::new(VALUES8.to_vec()).unwrap();
+    let expected_root = mt.root();
+
+    let ms = MerkleStore::from(&mt);
+
+    let path33 = ms.get_path(expected_root, NODE33).unwrap();
+
+    let pmt = PartialMerkleTree::with_paths([(3_u64, path33.value.into(), path33.path)]).unwrap();
+
+    assert_eq!(pmt.root(), expected_root.into());
+}
+
+#[test]
+fn add_and_get_paths() {
```

Comment:
I think this is more about testing `with_paths()` constructor rather than `add_path()`.